sdk/core/core-amqp/src/connectionConfig/connectionConfig.ts

```diff
@@ -91,7 +91,7 @@ export namespace ConnectionConfig {
     }
 
     if (!parsedCS.Endpoint.endsWith("/")) parsedCS.Endpoint += "/";
-
+    parsedCS.Endpoint = parsedCS.Endpoint.replace(/^\w+:\/\//, "sb://");
```

Comment:
The sole place where the updated endpoint is used in this function is to figure out what the "host" should be.
So, why not update the regex in the line that calculates the host instead of updating the endpoint here?

Reply:
> The sole place where the updated endpoint is used in this function is to figure out what the "host" should be.

Not just to calculate the host, "endpoint" is also a property of ConnectionConfig and this updated endpoint is being returned by this function.
![image](https://user-images.githubusercontent.com/10452642/78860379-ed5dae00-79e6-11ea-8d0a-23364d3348f8.png)

There are more places where the endpoint(not the host) is being used. Updating the endpoint right here would handle those cases too.

For example..
In service-bus
![image](https://user-images.githubusercontent.com/10452642/78860068-13368300-79e6-11ea-8b2e-449fcf65f0c4.png)

Reply:
Interesting... 
The screenshot is for creating the AMQP sender link
So, what would the service do when there is `https` instead of `sb` in the endpoint here?

Reply:
> The screenshot is for creating the AMQP sender link
So, what would the service do when there is https instead of sb in the endpoint here?

I'll look into that and get back.. but that isn't the only place. More references..
In event-hubs
![image](https://user-images.githubusercontent.com/10452642/78860019-ebdfb600-79e5-11ea-9363-474eeac03cbd.png)

In core-amqp
![image](https://user-images.githubusercontent.com/10452642/78859914-a15e3980-79e5-11ea-8847-16f3d78114ea.png)

Reply:
I am convinced of the change in this PR, but also curious as to how the service reacts :)

Reply:
![image](https://user-images.githubusercontent.com/10452642/79009997-c2be4300-7b15-11ea-8c3e-c101ed104654.png)
@ramya-rao-a 
Interestingly, it doesn't affect if it is `sb://` or `<anything>://`. I need to dig deeper to find out why.

Reply:
So, we are saying that if we removing our restriction that it should be `sb://` all the time then the service doesnt care at all and everything would just work?

Reply:
Yes, to my understanding. :)